packages/adapters/src/adapters/Cornerstone3D/Segmentation/generateSegmentation.ts

```diff
@@ -30,36 +54,66 @@ function generateSegmentation(images, labelmaps, metadata, options = {}) {
  * @returns The Seg derived dataSet.
  */
 function _createMultiframeSegmentationFromReferencedImages(
-    images,
-    metadata,
-    options
+  images,
+  metadata,
+  options
 ) {
-    const datasets = images.map(image => {
-        // add the sopClassUID to the dataset
-        const instance = metadata.get("instance", image.imageId);
-        return {
-            ...image,
-            ...instance,
-            // Todo: move to dcmjs tag style
-            SOPClassUID: instance.SopClassUID || instance.SOPClassUID,
-            SOPInstanceUID: instance.SopInstanceUID || instance.SOPInstanceUID,
-            PixelData: image.voxelManager.getScalarData(),
-            _vrMap: {
-                PixelData: "OW"
-            },
-            _meta: {}
-        };
-    });
+  const studyData = metadata.get(MetadataModules.STUDY_DATA, images[0].imageId);
+  const datasets = images.map((image) => {
+    const { imageId } = image;
+    const seriesData = metadata.get(MetadataModules.SERIES_DATA, imageId);
+    const imageData = metadata.get(MetadataModules.IMAGE_DATA, imageId);
+    return {
+      ...studyData,
+      ...seriesData,
+      ...imageData,
+      PixelData: image.voxelManager.getScalarData(),
+      // Declaring it as 16 bits allows the normalizer to work on 8 bit data
+      BitsAllocated: 16,
+      _vrMap: {
+        PixelData: 'OW',
+      },
+      _meta: {},
+    };
+  });
 
-    const multiframe = Normalizer.normalizeToDataset(datasets);
+  const isSingleNonMultiFrame =
+    datasets.length === 1 && !(datasets[0].NumberOfFrames > 1);
+  if (isSingleNonMultiFrame) {
+    // The normalizer doesn't handle just a single image correctly, but
+    // the segmentation version needs a multiframe object even for a single
+    // frame instance, so duplicate the object temporarily
+    datasets.push(datasets[0]);
+  }
+  // Directly use the SEGImageNormalizer to allow creating a SEG virtual instance consisting of
+  // whatever instance data we happen to have since this isn't actually creating
+  // a real multiframe, but rather a reference object
+  const normalizer = new SEGImageNormalizer(datasets);
+  normalizer.normalize();
+  const { dataset: multiframe } = normalizer;
 
-    if (!multiframe) {
-        throw new Error(
-            "Failed to normalize the multiframe dataset, the data is not multi-frame."
-        );
-    }
+  if (!multiframe) {
+    throw new Error(
+      'Failed to normalize the multiframe dataset, the data is not multi-frame.'
+    );
+  }
 
-    return new SegmentationDerivation([multiframe], options);
+  multiframe.SharedFunctionalGroupsSequence ||= {};
```

Comment:
Bunch of bugs caused by single frame data not being normalizeable by dcmjs, so these items fix the response.